material-slider/src/slider.ets

```diff
@@ -0,0 +1,193 @@
+          .onChange((value: number, mode: SliderChangeMode) => {
+            this.obj.setOutSetValue(value)
+          })
+        Row() {
+          Column() {
+            if (this.obj.getShowMin()) {
+              Text(this.obj.getMin().toFixed(0)).fontSize(16)
+            }
+          }.width('33%')
+
+          Column() {
+            if (this.obj.getShowValue()) {
+              Text(this.obj.getOutSetValue().toFixed(0)).fontSize(16)
+            }
+          }.width('33%')
+
+          Column() {
+            if (this.obj.getShowMax()) {
+              Text(this.obj.getMax().toFixed(0)).fontSize(16)
+            }
+          }.width('33%')
+        }
+      }
+      else if (this.obj.getSliderType() == "inset") {
+        Slider({
+          value: this.obj.getInSetValue(),
+          min: this.obj.getMin(),
+          max: this.obj.getMax(),
+          step: this.obj.getStep(),
+          direction: this.obj.getDirection(),
+          reverse: this.obj.getReverse(),
+          style: SliderStyle.InSet,
+        })
+          .blockColor(this.obj.getBlockColor())
+          .trackColor(this.obj.getTrackColor())
+          .selectedColor(this.obj.getSelectedColor())
+          .trackThickness(this.obj.getTrackThickness())
+          .showSteps(this.obj.getShowSteps())
+          .showTips(this.obj.getShowTips())
+          .onChange((value: number, mode: SliderChangeMode) => {
+            this.obj.setInSetValue(value)
+          })
+        Row() {
+          Column() {
+            if (this.obj.getShowMin()) {
+              Text(this.obj.getMin().toFixed(0)).fontSize(16)
+            }
+          }.width('33%')
+
+          Column() {
+            if (this.obj.getShowValue()) {
+              Text(this.obj.getInSetValue().toFixed(0)).fontSize(16)
+            }
+          }.width('33%')
+
+          Column() {
+            if (this.obj.getShowMax()) {
+              Text(this.obj.getMax().toFixed(0)).fontSize(16)
+            }
+          }.width('33%')
+        }
+      }
+    }
+  }
+}
+
+@Component
+export struct SliderDiscrete {
+  @State obj: SliderModel = new SliderModel(2, "sliderDiscrete")
+
+  build() {
+    Column({ space: 5 }) {
+      if (this.obj.getSliderType() == "outset") {
+        Slider({
+          value: this.obj.getOutSetValue(),
+          min: this.obj.getMin(),
+          max: this.obj.getMax(),
+          step: this.obj.getStep(),
+          direction: this.obj.getDirection(),
+          reverse: this.obj.getReverse(),
+          style: SliderStyle.OutSet
+        })
+          .blockColor(this.obj.getBlockColor())
+          .trackColor(this.obj.getTrackColor())
+          .selectedColor(this.obj.getSelectedColor())
+          .trackThickness(this.obj.getTrackThickness())
+          .showSteps(this.obj.getShowSteps())
+          .showTips(this.obj.getShowTips())
+          .onChange((value: number, mode: SliderChangeMode) => {
+            this.obj.setOutSetValue(value)
```

Comment:
create a callback function to send the updated values to the parent class (i.e the implementing class for this component)